js/src/Debug/Toolbar.vue

```diff
@@ -0,0 +1,445 @@
+                switchWidget(content_area.data('active-widget'), true);
+            }
+            // Move server global to the profile
+            if (ajax_request.server_global !== undefined) {
+                ajax_request.profile.globals['server'] = ajax_request.server_global;
+            }
+            // Move the data to either the get or post global depending on the request type
+            if (ajax_request.type === 'POST') {
+                ajax_request.profile.globals['post'] = ajax_request.data;
+            } else {
+                ajax_request.profile.globals['get'] = ajax_request.data;
+            }
+            refreshWidgetButtons();
+
+            if (reload_widget) {
+                // reload active widget
+                switchWidget(content_area.data('active-widget'), true);
+            }
+        });
+    }
+</script>
+
+<template>
+    <div id="debug-toolbar" class="position-fixed bottom-0 w-100 card" tabindex="0">
+        <div class="resize-handle mt-n2" @mousedown.prevent="$event.buttons === 1 && (is_dragging = true)"></div>
+        <div class="d-flex flex-row align-items-center">
+            <div class="debug-toolbar-badge d-flex">
+                <button type="button" class="btn btn-icon border-0 px-3 opacity-100 debug-logo" @click="showDebugToolbar()" disabled>
+                    <i class="ti ti-bug"></i>
+                </button>
+            </div>
+            <div class="debug-toolbar-content d-flex w-100 justify-content-between align-items-center">
+                <ul class="debug-toolbar-widgets nav nav-tabs align-items-center border-0" data-bs-toggle="tabs">
+                    <widget-button v-for="(widget) in getMainWidgets()" :id="widget.id" :title="widget.title" :icon="widget.icon"
+                        v-on:click="switchWidget(widget.id)" :active="widget.id === active_widget"
+                    ></widget-button>
+                </ul>
+                <div class="debug-toolbar-controls">
+                    <div class="debug-toolbar-control">
+                        <button type="button" class="btn btn-icon border-0 p-1" name="toggle_content_area" @click="toggleExtraContentArea()">
+                            <i class="ti ti-square-arrow-up"></i>
+                        </button>
+                        <button type="button" class="btn btn-icon border-0 p-1" title="Close" @click="hideDebugToolbar()">
+                            <i class="ti ti-square-x"></i>
+                        </button>
+                    </div>
+                </div>
+            </div>
+        </div>
+        <div id="debug-toolbar-expanded-content" class="w-100 card d-none pe-2">
+            <component v-if="active_widget" :is="active_widget_component" :initial_request="props.initial_request"
+               :ajax_requests="ajax_requests" @switchWidget="switchWidget" :widgets="widgets"></component>
+        </div>
+    </div>
+</template>
+
+<style scoped>
```

Comment:
These styles will be automatically prefixed with the generated component ID. Unless the selectors are wrapped inside a `:deep()`, they won't apply to child components. No need to worry about generic selectors leaking outside the scope of the feature.